include/net/net_pkt.h

```diff
@@ -2007,6 +2010,156 @@ const char *net_pkt_pool2str(struct net_buf_pool *pool);
 #define net_pkt_print(...)
 #endif /* CONFIG_NET_PKT_LOG_LEVEL >= LOG_LEVEL_DBG */
 
+/* New allocator, and API are defined below.
+ * This will be simpler when time will come to get rid of former API above.
+ */
+#if CONFIG_NET_PKT_LOG_LEVEL >= LOG_LEVEL_DBG
+
+struct net_pkt *net_pkt_allocate_debug(s32_t timeout,
+				       const char *caller, int line);
+#define net_pkt_allocate(_timeout)					\
+	net_pkt_allocate_debug(_timeout, __func__, __LINE__)
+
+struct net_pkt *net_pkt_rx_allocate_debug(s32_t timeout,
+					  const char *caller, int line);
+#define net_pkt_rx_allocate(_timeout)				\
+	net_pkt_rx_allocate_debug(_timeout, __func__, __LINE__)
+
+struct net_pkt *net_pkt_allocate_on_iface_debug(struct net_if *iface,
+						s32_t timeout,
+						const char *caller,
+						int line);
+#define net_pkt_allocate_on_iface_(_iface, _timeout)			\
+	net_pkt_allocate_on_iface_debug(_iface, _timeout, __func__, __LINE__)
+
+struct net_pkt *net_pkt_rx_allocate_on_iface_debug(struct net_if *iface,
+						   s32_t timeout,
+						   const char *caller,
+						   int line);
+#define net_pkt_rx_allocate_on_iface_(_iface, _timeout)			\
+	net_pkt_rx_allocate_on_iface_debug(_iface, _timeout,		\
+					   __func__, __LINE__)
+
+int net_pkt_allocate_buffer_debug(struct net_pkt *pkt,
+				  size_t size,
+				  enum net_ip_protocol proto,
+				  s32_t timeout,
+				  const char *caller, int line);
+#define net_pkt_allocate_buffer(_pkt, _size, _proto, _timeout)		\
+	net_pkt_allocate_buffer_debug(_pkt, _size, _proto, _timeout,	\
+				      __func__, __LINE__)
+
+struct net_pkt *net_pkt_allocate_with_buffer_debug(struct net_if *iface,
+						   size_t size,
+						   sa_family_t family,
+						   enum net_ip_protocol proto,
+						   s32_t timeout,
+						   const char *caller,
+						   int line);
+#define net_pkt_allocate_with_buffer(_iface, _size, _family,		\
+				     _proto, _timeout)			\
+	net_pkt_allocate_with_buffer_debug(_iface, _size, _family,	\
+					   _proto, _timeout,		\
+					   __func__, __LINE__)
+
+struct net_pkt *net_pkt_rx_allocate_with_buffer_debug(
+						struct net_if *iface,
+						size_t size,
+						sa_family_t family,
+						enum net_ip_protocol proto,
+						s32_t timeout,
+						const char *caller,
+						int line);
+#define net_pkt_rx_allocate_with_buffer(_iface, _size, _family,		\
+					_proto, _timeout)		\
+	net_pkt_rx_allocate_with_buffer_debug(_iface, _size, _family,	\
+					      _proto, _timeout,		\
+					      __func__, __LINE__)
+#else
+
+/**
+ * @brief Allocate an initialized net_pkt
+ *
+ * Note: for the time being, 2 pools are used. One for TX and one for RX.
+ *       This allocater has to be used for TX.
+ *
+ * @param timeout Maximum time in milliseconds to wait for an allocation.
+ *
+ * @return a pointer to a newly allocated net_pkt on success, NULL otherwise.
+ */
+struct net_pkt *net_pkt_allocate(s32_t timeout);
```

Comment:
Shouldn't we call this net_pkt_tx_allocate() in order to follow the same syntax as the rx allocator one?

Reply:
the rx stuff is an in-between name. Idea would be to have only 1 allocator for all pkt (thus one slab, etc...). However, we will still be able to know (for stats, net mem in shell etc...) if a pkt was allocated for tx or rx, there will be a bit in net_pkt to give that info and so on (net_recv_data will set it for rx, tx will be default). That's not going to be part of that PR though,